tools/devops/automation/templates/build/publish-html.yml

```diff
@@ -118,7 +118,82 @@ steps:
       $apiDiffData | ConvertTo-Json | Write-Host
       Write-Host "##vso[task.setvariable variable=APIDIFF_JSON_PATH]$path"
     displayName: 'Create API from stable diff gists'
-    timeoutInMinutes: 1
+    timeoutInMinutes: 5
+    env:
+      GITHUB_TOKEN: $(GitHub.Token)
+
+- ${{ if eq(parameters.runTests, true) }}:
+  - pwsh: |
+      Import-Module $Env:SYSTEM_DEFAULTWORKINGDIRECTORY\xamarin-macios\tools\devops\automation\scripts\GitHub.psm1
+
+      $apiDiffRoot = "$Env:STABLE_APID_GENERATOR_DIFF_PATH"
+      $filePatterns = @{
+        "iOS" = "^ios-[a-zA-Z0-9\-]*\.md$";
+        "macOS" = "^macos-[a-zA-Z0-9\-]*\.md$";
+        "tvOS" = "^tvos-[a-zA-Z0-9\-]*\.md$";
+        "watchOS" = "^watchos-[a-zA-Z0-9\-]*\.md$";
+        "dotnet-iOS" = "^dotnet-ios-[a-zA-Z0-9\-]*\.md$";
+        "dotnet-tvOS" = "^dotnet-tvos-[a-zA-Z0-9\-]*\.md$";
+        "dotnet-MacCatalyst" = "^dotnet-maccatalyst-[a-zA-Z0-9\-]*\.md$";
+        "dotnet-macOS" = "^dotnet-macos-[a-zA-Z0-9\-]*\.md$";
+        "dotnet-legacy-iOS" = "^dotnet-legacy-ios-[a-zA-Z0-9\-]*\.md$";
+        "dotnet-legacy-tvOS" = "^dotnet-legacy-tvos-[a-zA-Z0-9\-]*\.md$";
+        "dotnet-legacy-macOS" = "^dotnet-legacy-macos-[a-zA-Z0-9\-]*\.md$";
+        "dotnet-macCatiOS" = "^dotnet-maccatios-[a-zA-Z0-9\-]*\.md$";
+      }
+
+      [System.Collections.Generic.List[System.Object]]$gistsObj = @()
+      $gists = @{}
+
+      foreach ($key in $filePatterns.Keys) {
+        $filter = $filePatterns[$key]
+        $fileName = Get-ChildItem $apiDiffRoot -Name | Where-Object{$_ -match $filter}
+        if ($fileName) {
+          $obj = New-GistObjectDefinition -Name $fileName -Path "$apiDiffRoot\$fileName" -Type "markdown"
```

Comment:
The call can return none when it fails. I had to fix that already due to a big we had, please do the same as in PR https://github.com/xamarin/xamarin-macios/pull/13483

Reply:
Good catch!